.pa11yci.js

```diff
@@ -1,41 +1,84 @@
-const viewports = [
-	{
-		width: 768,
-		height: 1024
-	},
-	{
-		width: 490,
-		height: 732
-	},
-	{
-		width: 320,
-		height: 480
-	}
-];
-
-const urls = [
-	`https://n-ui.ft.com/n-ui/test-page/${process.env.CIRCLE_BUILD_NUM}/test-page.html`
-]
+const chalk = require('chalk');
+const fs = require('fs');
+const path = require('path');
+const error = chalk.bold.red;
 
 const config = {
-	defaults: {
-		page: {
-			headers: {
-				"Cookie": "next-flags=ads:off,cookieMessage:off,nTeaserStreamPage:on; secure=true"
-			}
-		},
-		timeout: 25000
+	"defaults": {
+		"timeout": 10000,
+		"rootElement": "#a11y"
 	},
-	urls: []
+	"urls": []
+};
+
+const pa11yIgnore = [
+	'.git',
+	'node_modules',
+	'bower_components',
+	'public',
+	'_deploy',
+	'_sass-utils',
+	'_test-server',
+	'ads',
+	'buttons',
+	'colors',
+	'cookie-message',
+	'date',
+	'dist',
+	'expander',
+	'forms',
+	'grid',
+	'icons',
+	'js-setup',
+	'layout',
+	'myft',
+	'myft-common',
+	'myft-digest-promo',
+	'myft-hint',
+	'n-ui',
+	'normalize',
+	'notification',
+	'opt-out',
+	'overlay',
+	'page-heading',
+	'subscription-offer-prompt',
+	'test',
+	'tour-tip',
+	'tour-tip-group',
+	'tracking',
+	'typography',
+	'utils',
+	'viewport',
+	'welcome-message'
+];
+
+function getDirectories(srcpath) {
+	return fs.readdirSync(srcpath).filter(function(file) {
+		return fs.statSync(path.join(srcpath, file)).isDirectory() && pa11yIgnore.indexOf(file) < 0;
+	});
 }
 
-for (viewport of viewports) {
-	for (url of urls) {
+const components = getDirectories('./')
+const missingPa11yConfig = [];
+
+components.forEach((component) => {
+	const hasPa11yConfig = fs.readdirSync(`./${component}`).some((file) => {
+		return file === 'pa11y-config.js';
+	});
+
+	if (!hasPa11yConfig) {
+		missingPa11yConfig.push(component);
+	}
+
+	if(hasPa11yConfig) {
 		config.urls.push({
-			url: url,
-			viewport: viewport
-		})
+			"url": `localhost:5005/components/${component}`
+		});
 	}
+});
+
+if(missingPa11yConfig.length) {
+	throw new Error(error(`Error: Components need to have a pa11y-config, components missing a pa11y-config: ${missingPa11yConfig.join()}`));
```

Comment:
I made this build fail purposefully to demo what will happen

https://circleci.com/gh/Financial-Times/n-ui/3072

I think it should also point to the readme which should explain how to set this up and in small print how to remove a component if really necessary